net/dns/resolver/forwarder.go

```diff
@@ -379,6 +380,12 @@ func (c *fwdConn) send(packet []byte, dst netaddr.IPPort) {
 			backOff(err)
 			continue
 		}
+		if errors.Is(err, syscall.EHOSTUNREACH) {
```

Comment:
clarify how this differents from line 394 just below?

Reply:
Wow there's a lot of subtle difference there. Yeah, that seems like a good idea.